From 6ed7312e533d70d2f2f2b915f78e0373db824b31 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Thu, 19 Apr 2012 15:46:40 +0800
Subject: [PATCH 315/505] ENGR00180239-1: Enable text format output for unit
 test documentation

The generated ut_um.text will be included in daily-built rootfs.
Thus users can easily run unit test by referring to it natively.

Note that VI command in rootfs is a simple version from busybox, which does
not inteprete the text file (with table format for each test inside) very well
as it does on Host PC. Recommend use `cat' command (saying, 'cat ut_um.text').

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 test/doc/Makefile | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/test/doc/Makefile b/test/doc/Makefile
index 15b4ac6..b09d207 100644
--- a/test/doc/Makefile
+++ b/test/doc/Makefile
@@ -1,17 +1,17 @@
 DOCMAIN :=$(wildcard *.toc)
 
-all: pre-build chunked single
+all: pre-build single text chunked
 
 pre-build:
 	./gen_conf.sh $(PLATFORM)
 
-chunked: $(DOCMAIN)
+single: $(DOCMAIN)
 	asciidoc -a toc $<
 ifneq ($(OBJDIR),)
 	cp -f *.html $(OBJDIR)
 endif
 
-single:
+chunked:
 
 pdf: $(DOCMAIN)
 	a2x -dbook -f pdf $<
@@ -19,6 +19,12 @@ ifneq ($(OBJDIR),)
 	cp -f *.pdf $(OBJDIR)
 endif
 
+text: $(DOCMAIN)
+	a2x -dbook -f text $<
+ifneq ($(OBJDIR),)
+	cp -f *.text $(OBJDIR)
+endif
+
 .PHONY: clean
 clean:
 	rm -f *.pdf *.html
-- 
1.8.0

